src/net.cpp

```diff
@@ -1747,18 +1767,19 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         bool anchor = false;
         bool fFeeler = false;
 
-        // Determine what type of connection to open. Opening
-        // BLOCK_RELAY connections to addresses from anchors.dat gets the highest
-        // priority. Then we open OUTBOUND_FULL_RELAY priority until we
-        // meet our full-relay capacity. Then we open BLOCK_RELAY connection
-        // until we hit our block-relay-only peer limit.
-        // GetTryNewOutboundPeer() gets set when a stale tip is detected, so we
-        // try opening an additional OUTBOUND_FULL_RELAY connection. If none of
-        // these conditions are met, check to see if it's time to try an extra
-        // block-relay-only peer (to confirm our tip is current, see below) or the next_feeler
-        // timer to decide if we should open a FEELER.
-
-        if (!m_anchors.empty() && (nOutboundBlockRelay < m_max_outbound_block_relay)) {
+        // Determine what type of connection to open, in order of priority:
+        // * SENSITIVE_RELAY if needed
+        // * BLOCK_RELAY connections to addresses from anchors.dat
+        // * OUTBOUND_FULL_RELAY until we reach full-relay capacity
+        // * BLOCK_RELAY until we reach block-relay-only peer limit.
+        // * OUTBOUND_FULL_RELAY if GetTryNewOutboundPeer() is true (a stale tip is detected)
+        // * BLOCK_RELAY if it's time to try an extra block-relay-only peer (to confirm our tip is current)
+        // * FEELER if it's time to try a feeler
+        // * else retry the loop (sleep a bit and start from the top of this list)
+
+        if (open_sensitive_relay) {
```

Comment:
I suppose it bypass the `maxconnection`, does it? If so, it worths to update the `-maxconnections` mentioning it.

Reply:
Yes. That is a bit hidden/implicit in the creation of the grant, in `master`:

```cpp
CSemaphoreGrant grant(*semOutbound);
```

this would hang if there are too many connections, waiting for a free slot to be available (somebody to disconnect).

Added a mention to it in the description of `-maxconnections`.